cli/cli.go

```diff
@@ -135,11 +134,20 @@ func (c *CLI) validateArgs(ctx context.Context, args []string) {
 }
 
 // unmarshalRequest reads input from std.in and unmarshal it into given request struct
-func (c *CLI) unmarshalRequest(request any) error {
+func (c *CLI) unmarshalRequest(request plugin.Request) error {
 	if err := json.NewDecoder(os.Stdin).Decode(request); err != nil {
 		c.logger.Errorf("%s unmarshalling error :%v", reflect.TypeOf(request), err)
 		return plugin.NewJSONParsingError(plugin.ErrorMsgMalformedInput)
 	}
+
+	if err := request.Validate(); err != nil {
+		c.logger.Errorf("%s validation error :%v", reflect.TypeOf(request), err)
+		if e, ok := err.(*plugin.Error); ok {
```

Comment:
Do we need to check this? Because errors from `request.Validate()` is always `nil` or `*plugin.Error`.

Reply:
@priteshbandi I couldn't see the changes though, you might forget to push them?